Warning on HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression

Build on macOS and remote Heroku produces a series of warnings during compilation.


### Steps to reproduce
swift build on Mac, and vapor/vapor build pack for heroku

### Expected behavior
No warnings.
### Actual behavior

macOS:
$ swift build
```
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
/Users/user/web-project/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
                                     ^
```

Heroku build:
```
remote: [909/910] Compiling Vapor Application.swift
remote: /tmp/build_aaad98dcb6bb41b3137bde2fa3c158a0/.build/checkouts/vapor/Sources/Vapor/HTTP/Server/HTTPServer.swift:251:38: warning: no calls to throwing functions occur within 'try' expression
remote:                         tlsHandler = try NIOSSLServerHandler(context: sslContext)
remote:                                      ^
```
### Environment
Swift 5.2.4
Using vapor-beta, and it doesn't have a --version

* Vapor Framework version:

        "package": "vapor",
        "repositoryURL": "https://github.com/vapor/vapor.git",
        "state": {
          "branch": null,
          "revision": "dc2aa1e02e04a47b67cb0dabed628fe844900f30",
          "version": "4.14.0"
        }

* OS version:
macOS 10.5.5
linux Heroku remote
```
~ $ uname -r
4.4.0-1073-aws
~ $ cat /etc/os-release 
NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.4 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```

--- a/Package.swift
+++ b/Package.swift
@@ -34,7 +34,7 @@ let package = Package(
         .package(url: "https://github.com/apple/swift-nio.git", from: "2.13.1"),
         
         // Bindings to OpenSSL-compatible libraries for TLS support in SwiftNIO
-        .package(url: "https://github.com/apple/swift-nio-ssl.git", from: "2.4.1"),
+        .package(url: "https://github.com/apple/swift-nio-ssl.git", from: "2.8.0"),
         
         // HTTP/2 support for SwiftNIO
         .package(url: "https://github.com/apple/swift-nio-http2.git", from: "1.11.0"),

--- a/Sources/Vapor/HTTP/Server/HTTPServer.swift
+++ b/Sources/Vapor/HTTP/Server/HTTPServer.swift
@@ -248,7 +248,7 @@ private final class HTTPServerConnection {
                     let tlsHandler: NIOSSLServerHandler
                     do {
                         sslContext = try NIOSSLContext(configuration: tlsConfiguration)
-                        tlsHandler = try NIOSSLServerHandler(context: sslContext)
+                        tlsHandler = NIOSSLServerHandler(context: sslContext)
                     } catch {
                         configuration.logger.error("Could not configure TLS: \(error)")
                         return channel.close(mode: .all)

